File: src/models/approval.py
from __future__ import annotations

from datetime import datetime
from enum import Enum

from pydantic import BaseModel
from pydantic import Field


class ApprovalDecision(str, Enum):
    APPROVE = "approve"
    REJECT = "reject"
    INVESTIGATE = "investigate"


class ApprovalRecord(BaseModel):
    investigation_id: str = Field(min_length=1)
    decision: ApprovalDecision
    decided_by: str = Field(min_length=1)
    decided_at: datetime
    reason: str | None = None


ApprovalStore = dict[str, ApprovalRecord]

File: dotnet/src/IncidentWarRoom.Core/Models/Approval.cs
using System;

namespace IncidentWarRoom.Core.Models;

public enum ApprovalDecision
{
    Approve,
    Reject,
    Investigate,
}

public sealed record ApprovalRecord(
    string InvestigationId,
    ApprovalDecision Decision,
    string DecidedBy,
    DateTimeOffset DecidedAt,
    string? Reason
);
